common/check_expired_dids

```diff
@@ -1,59 +1,58 @@
-#!/usr/bin/env python
-# Copyright European Organization for Nuclear Research (CERN) 2013
+#!/usr/bin/env python3
+# Copyright European Organization for Nuclear Research (CERN) since 2012
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
-# You may not use this file except in compliance with the License.
-# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
 #
-# Authors:
-# - Vincent Garonne, <vincent.garonne@cern.ch>, 2013
-# - Thomas Beermann, <thomas.beermann@cern.ch>, 2019
-# - Eric Vaandering <ewv@fnal.gov>, 2020-2021
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
 
 """
 Probe to check the backlog of expired dids.
 """
-from __future__ import print_function
 
 import sys
 import traceback
 from datetime import datetime
 
-from prometheus_client import CollectorRegistry, Gauge, push_to_gateway
-from rucio.common.config import config_get
+from sqlalchemy import and_, func, null, select
+
 from rucio.db.sqla import models
 from rucio.db.sqla.session import get_session
-from rucio.db.sqla.util import get_count
 
-from utils.common import probe_metrics
+from utils.common import PrometheusPusher
 
 # Exit statuses
 OK, WARNING, CRITICAL, UNKNOWN = 0, 1, 2, 3
 
-PROM_SERVERS = config_get('monitor', 'prometheus_servers', raise_exception=False, default='')
-if PROM_SERVERS != '':
-    PROM_SERVERS = PROM_SERVERS.split(',')
-
 if __name__ == "__main__":
     try:
-        registry = CollectorRegistry()
         session = get_session()
-        query = session.query(models.DataIdentifier.scope).filter(models.DataIdentifier.expired_at.isnot(None),
-                                                                  models.DataIdentifier.expired_at < datetime.utcnow())
-        result = get_count(query)
-        # Possible check against a threshold. If result > max_value then sys.exit(CRITICAL)
-        probe_metrics.gauge(name='undertaker.expired_dids').set(result)
-        Gauge('undertaker_expired_dids', '', registry=registry).set(result)
-
-        if len(PROM_SERVERS):
-            for server in PROM_SERVERS:
-                try:
-                    push_to_gateway(server.strip(), job='check_expired_dids', registry=registry)
-                except:
-                    continue
-
-        print(result)
-    except:
+        with PrometheusPusher() as manager:
```

Comment:
I wonder if this should be moved further down below and only encompass `manager.gauge()`. Isn’t it preferable for a context to be as small as possible?

I do not have a particularly strong opinion on this, but the benefits are (1) related code is closer together and (2) unrelated code does not have to be indented by one additional level.

Reply:
I don't really have a preference, I'm just following the convention from the other code using the PrometheusPusher. Not really a reason to keep it that way though.